pkg/virt-controller/watch/vmi.go

```diff
@@ -1223,6 +1188,56 @@ func (c *VMIController) getHotplugVolumes(vmi *virtv1.VirtualMachineInstance, vi
 	return hotplugVolumes
 }
 
+func (c *VMIController) cleanupWaitForFirstConsumerTemporaryPods(vmi *virtv1.VirtualMachineInstance, virtlauncherPod *k8sv1.Pod) error {
+	// Get all pods from the namespace
+	pods, err := c.listPodsFromNamespace(virtlauncherPod.Namespace)
+	if err != nil {
+		return err
+	}
```

Comment:
if we're goign to list all pods anyway, is there any reason to treat one pod `virtlauncherpod` any different than the rest? the virtlauncherpod is just another temp pod, right?

could this function be as simple as listing all trigger pods and deleting any that are running/succeeded/failed?

Reply:
Now that I think about that, no, all I care about is this a temp pod or not. Doesn't matter if it is an attachment pod or virt launcher pod.